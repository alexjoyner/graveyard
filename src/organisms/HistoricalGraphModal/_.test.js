import React from 'react';
import { shallow, mount } from 'enzyme';
import { BaseHistoricalGraphModal } from './';

describe('BaseHistoricalGraphModal component', () => {
  describe('Empty Hidden Chart Component', () => {
    let component;
    beforeEach(() => {
      const props = {
        startLoadingNotif: jest.fn(),
        stopLoadingNotif: jest.fn(),
        dispatchNewHistoricalData: jest.fn(),
        showHistoryModal: jest.fn(),
        closeHistoryModal: jest.fn(),
        modalStage: 'CLOSED',
        chartPoints: {},
      };
      component = shallow(<BaseHistoricalGraphModal {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('Shown Chart Component', () => {
    let component;
    let props;
    beforeEach(() => {
      props = {
        startLoadingNotif: jest.fn(),
        stopLoadingNotif: jest.fn(),
        dispatchNewHistoricalData: jest.fn(),
        showHistoryModal: jest.fn(),
        closeHistoryModal: jest.fn(),
        modalStage: 'SHOWN',
        chartPoints: {},
      };
      component = shallow(<BaseHistoricalGraphModal {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
    it('should allow closing history modal', () => {
      component.find('Button').last().props().onClick();
      expect(props.closeHistoryModal).toHaveBeenCalledTimes(1);
    });
  });
  describe('Full Mounted Component', () => {
    let props;
    let spy;
    beforeEach(() => {
      props = {
        startLoadingNotif: jest.fn(),
        stopLoadingNotif: jest.fn(),
        dispatchNewHistoricalData: jest.fn(),
        showHistoryModal: jest.fn(),
        closeHistoryModal: jest.fn(),
        modalStage: 'SHOWN',
        chartPoints: {},
      };
    });
    afterEach(() => {
      if (spy) {
        spy.mockClear();
      }
    });
    it('Should render without exploding', () => {
      const component = mount(<BaseHistoricalGraphModal {...props} />);
      expect(component).toBeDefined();
    });
    it('Should call component did update', () => {
      spy = jest.spyOn(BaseHistoricalGraphModal.prototype, 'componentDidUpdate');
      const component = mount(<BaseHistoricalGraphModal {...props} />);
      component.setProps({ ...props, modalStage: 'TESTING' });
      expect(spy).toHaveBeenCalled();
    });
    it('Should call runBuildGraphProcess', () => {
      spy = jest.spyOn(BaseHistoricalGraphModal.prototype, 'runBuildGraphProcess');
      const component = mount(<BaseHistoricalGraphModal {...props} />);
      component.setProps({ ...props, modalStage: 'BUILDING' });
      expect(spy).toHaveBeenCalled();
    });
    it('Should call runBuildGraphProcess from IntervalButton', () => {
      spy = jest.spyOn(BaseHistoricalGraphModal.prototype, 'runBuildGraphProcess');
      const component = mount(<BaseHistoricalGraphModal {...props} />);
      const IntervalButton = component.find('IntervalButton');
      IntervalButton.forEach((Button) => {
        Button.props().onClick();
      });
      expect(spy).toHaveBeenCalledTimes(5);
    });
  });
});
