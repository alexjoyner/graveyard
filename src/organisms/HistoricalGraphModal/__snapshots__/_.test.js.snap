// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseHistoricalGraphModal component Empty Hidden Chart Component Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseHistoricalGraphModal
    chartPoints={Object {}}
    closeHistoryModal={[MockFunction]}
    dispatchNewHistoricalData={[MockFunction]}
    modalData={Array []}
    modalStage="CLOSED"
    showHistoryModal={[MockFunction]}
    startLoadingNotif={[MockFunction]}
    stopLoadingNotif={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "style": Object {
        "visibility": "hidden",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "style": Object {
          "visibility": "hidden",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`BaseHistoricalGraphModal component Shown Chart Component Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseHistoricalGraphModal
    chartPoints={Object {}}
    closeHistoryModal={[MockFunction]}
    dispatchNewHistoricalData={[MockFunction]}
    modalData={Array []}
    modalStage="SHOWN"
    showHistoryModal={[MockFunction]}
    startLoadingNotif={[MockFunction]}
    stopLoadingNotif={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Panel
          width="90%"
        >
          <IntervalButton
            color="primary"
            onClick={[Function]}
            type="oneDay"
          >
            1 Day
          </IntervalButton>
          <IntervalButton
            color="primary"
            onClick={[Function]}
            type="oneWeek"
          >
            1 Week
          </IntervalButton>
          <IntervalButton
            color="primary"
            onClick={[Function]}
            type="oneMonth"
          >
            1 Month
          </IntervalButton>
          <IntervalButton
            color="primary"
            onClick={[Function]}
            type="sixMonths"
          >
            6 Months
          </IntervalButton>
          <IntervalButton
            color="primary"
            onClick={[Function]}
          >
            All
          </IntervalButton>
        </Panel>,
        <Button
          color="primary"
          ghost={false}
          onClick={[Function]}
          size="default"
        >
          <span>
            Close
          </span>
        </Button>,
        <RoHighChart
          config={
            Object {
              "boost": Object {
                "useGPUTranslations": true,
              },
              "chart": Object {
                "type": "line",
                "zoomType": "x",
              },
              "legend": Object {
                "enabled": true,
              },
              "series": Array [],
              "subtitle": Object {
                "text": "Click and drag in the plot area to zoom in",
              },
              "title": Object {
                "text": "Historical Data View",
              },
              "xAxis": Object {
                "labels": Object {
                  "style": Object {
                    "color": "white",
                  },
                },
                "type": "datetime",
              },
              "yAxis": Object {
                "labels": Object {
                  "format": "{value}",
                  "style": Object {
                    "color": "white",
                  },
                },
                "title": Object {
                  "text": "Value",
                },
              },
            }
          }
          theme={
            Object {
              "chart": Object {
                "backgroundColor": "#2d3436",
                "borderColor": "#ffffff",
                "plotBackgroundColor": "#2d3436",
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "12px",
                },
                "type": "line",
              },
              "colors": Array [
                "#fdcb6e",
                "#ff7675",
                "#55efc4",
                "#74b9ff",
              ],
              "credits": Object {
                "enabled": false,
              },
              "exporting": Object {
                "enabled": true,
              },
              "legend": Object {
                "itemStyle": Object {
                  "color": "white",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "10px",
                },
              },
              "subtitle": Object {
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "18px",
                },
                "text": "",
              },
              "title": Object {
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "18px",
                },
                "text": "Categorized, four series",
              },
              "tooltip": Object {
                "shared": true,
              },
              "yAxis": Object {
                "labels": Object {},
                "title": Object {
                  "style": Object {
                    "color": "#ffffff",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "12px",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                },
              },
            }
          }
        />,
      ],
      "height": "auto",
      "width": "90%",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneDay"
            >
              1 Day
            </IntervalButton>,
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneWeek"
            >
              1 Week
            </IntervalButton>,
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneMonth"
            >
              1 Month
            </IntervalButton>,
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="sixMonths"
            >
              6 Months
            </IntervalButton>,
            <IntervalButton
              color="primary"
              onClick={[Function]}
            >
              All
            </IntervalButton>,
          ],
          "width": "90%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1 Day",
              "color": "primary",
              "onClick": [Function],
              "type": "oneDay",
            },
            "ref": null,
            "rendered": "1 Day",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1 Week",
              "color": "primary",
              "onClick": [Function],
              "type": "oneWeek",
            },
            "ref": null,
            "rendered": "1 Week",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1 Month",
              "color": "primary",
              "onClick": [Function],
              "type": "oneMonth",
            },
            "ref": null,
            "rendered": "1 Month",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "6 Months",
              "color": "primary",
              "onClick": [Function],
              "type": "sixMonths",
            },
            "ref": null,
            "rendered": "6 Months",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "All",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "All",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            Close
          </span>,
          "color": "primary",
          "ghost": false,
          "onClick": [Function],
          "size": "default",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Close",
          },
          "ref": null,
          "rendered": "Close",
          "type": "span",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "config": Object {
            "boost": Object {
              "useGPUTranslations": true,
            },
            "chart": Object {
              "type": "line",
              "zoomType": "x",
            },
            "legend": Object {
              "enabled": true,
            },
            "series": Array [],
            "subtitle": Object {
              "text": "Click and drag in the plot area to zoom in",
            },
            "title": Object {
              "text": "Historical Data View",
            },
            "xAxis": Object {
              "labels": Object {
                "style": Object {
                  "color": "white",
                },
              },
              "type": "datetime",
            },
            "yAxis": Object {
              "labels": Object {
                "format": "{value}",
                "style": Object {
                  "color": "white",
                },
              },
              "title": Object {
                "text": "Value",
              },
            },
          },
          "theme": Object {
            "chart": Object {
              "backgroundColor": "#2d3436",
              "borderColor": "#ffffff",
              "plotBackgroundColor": "#2d3436",
              "style": Object {
                "color": "#dfe6e9",
                "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                "fontSize": "12px",
              },
              "type": "line",
            },
            "colors": Array [
              "#fdcb6e",
              "#ff7675",
              "#55efc4",
              "#74b9ff",
            ],
            "credits": Object {
              "enabled": false,
            },
            "exporting": Object {
              "enabled": true,
            },
            "legend": Object {
              "itemStyle": Object {
                "color": "white",
                "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                "fontSize": "10px",
              },
            },
            "subtitle": Object {
              "style": Object {
                "color": "#dfe6e9",
                "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                "fontSize": "18px",
              },
              "text": "",
            },
            "title": Object {
              "style": Object {
                "color": "#dfe6e9",
                "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                "fontSize": "18px",
              },
              "text": "Categorized, four series",
            },
            "tooltip": Object {
              "shared": true,
            },
            "yAxis": Object {
              "labels": Object {},
              "title": Object {
                "style": Object {
                  "color": "#ffffff",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "12px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              },
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Panel
            width="90%"
          >
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneDay"
            >
              1 Day
            </IntervalButton>
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneWeek"
            >
              1 Week
            </IntervalButton>
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="oneMonth"
            >
              1 Month
            </IntervalButton>
            <IntervalButton
              color="primary"
              onClick={[Function]}
              type="sixMonths"
            >
              6 Months
            </IntervalButton>
            <IntervalButton
              color="primary"
              onClick={[Function]}
            >
              All
            </IntervalButton>
          </Panel>,
          <Button
            color="primary"
            ghost={false}
            onClick={[Function]}
            size="default"
          >
            <span>
              Close
            </span>
          </Button>,
          <RoHighChart
            config={
              Object {
                "boost": Object {
                  "useGPUTranslations": true,
                },
                "chart": Object {
                  "type": "line",
                  "zoomType": "x",
                },
                "legend": Object {
                  "enabled": true,
                },
                "series": Array [],
                "subtitle": Object {
                  "text": "Click and drag in the plot area to zoom in",
                },
                "title": Object {
                  "text": "Historical Data View",
                },
                "xAxis": Object {
                  "labels": Object {
                    "style": Object {
                      "color": "white",
                    },
                  },
                  "type": "datetime",
                },
                "yAxis": Object {
                  "labels": Object {
                    "format": "{value}",
                    "style": Object {
                      "color": "white",
                    },
                  },
                  "title": Object {
                    "text": "Value",
                  },
                },
              }
            }
            theme={
              Object {
                "chart": Object {
                  "backgroundColor": "#2d3436",
                  "borderColor": "#ffffff",
                  "plotBackgroundColor": "#2d3436",
                  "style": Object {
                    "color": "#dfe6e9",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "12px",
                  },
                  "type": "line",
                },
                "colors": Array [
                  "#fdcb6e",
                  "#ff7675",
                  "#55efc4",
                  "#74b9ff",
                ],
                "credits": Object {
                  "enabled": false,
                },
                "exporting": Object {
                  "enabled": true,
                },
                "legend": Object {
                  "itemStyle": Object {
                    "color": "white",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "10px",
                  },
                },
                "subtitle": Object {
                  "style": Object {
                    "color": "#dfe6e9",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "18px",
                  },
                  "text": "",
                },
                "title": Object {
                  "style": Object {
                    "color": "#dfe6e9",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "18px",
                  },
                  "text": "Categorized, four series",
                },
                "tooltip": Object {
                  "shared": true,
                },
                "yAxis": Object {
                  "labels": Object {},
                  "title": Object {
                    "style": Object {
                      "color": "#ffffff",
                      "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                      "fontSize": "12px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  },
                },
              }
            }
          />,
        ],
        "height": "auto",
        "width": "90%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <IntervalButton
                color="primary"
                onClick={[Function]}
                type="oneDay"
              >
                1 Day
              </IntervalButton>,
              <IntervalButton
                color="primary"
                onClick={[Function]}
                type="oneWeek"
              >
                1 Week
              </IntervalButton>,
              <IntervalButton
                color="primary"
                onClick={[Function]}
                type="oneMonth"
              >
                1 Month
              </IntervalButton>,
              <IntervalButton
                color="primary"
                onClick={[Function]}
                type="sixMonths"
              >
                6 Months
              </IntervalButton>,
              <IntervalButton
                color="primary"
                onClick={[Function]}
              >
                All
              </IntervalButton>,
            ],
            "width": "90%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1 Day",
                "color": "primary",
                "onClick": [Function],
                "type": "oneDay",
              },
              "ref": null,
              "rendered": "1 Day",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1 Week",
                "color": "primary",
                "onClick": [Function],
                "type": "oneWeek",
              },
              "ref": null,
              "rendered": "1 Week",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1 Month",
                "color": "primary",
                "onClick": [Function],
                "type": "oneMonth",
              },
              "ref": null,
              "rendered": "1 Month",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "6 Months",
                "color": "primary",
                "onClick": [Function],
                "type": "sixMonths",
              },
              "ref": null,
              "rendered": "6 Months",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "All",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "All",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              Close
            </span>,
            "color": "primary",
            "ghost": false,
            "onClick": [Function],
            "size": "default",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Close",
            },
            "ref": null,
            "rendered": "Close",
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "config": Object {
              "boost": Object {
                "useGPUTranslations": true,
              },
              "chart": Object {
                "type": "line",
                "zoomType": "x",
              },
              "legend": Object {
                "enabled": true,
              },
              "series": Array [],
              "subtitle": Object {
                "text": "Click and drag in the plot area to zoom in",
              },
              "title": Object {
                "text": "Historical Data View",
              },
              "xAxis": Object {
                "labels": Object {
                  "style": Object {
                    "color": "white",
                  },
                },
                "type": "datetime",
              },
              "yAxis": Object {
                "labels": Object {
                  "format": "{value}",
                  "style": Object {
                    "color": "white",
                  },
                },
                "title": Object {
                  "text": "Value",
                },
              },
            },
            "theme": Object {
              "chart": Object {
                "backgroundColor": "#2d3436",
                "borderColor": "#ffffff",
                "plotBackgroundColor": "#2d3436",
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "12px",
                },
                "type": "line",
              },
              "colors": Array [
                "#fdcb6e",
                "#ff7675",
                "#55efc4",
                "#74b9ff",
              ],
              "credits": Object {
                "enabled": false,
              },
              "exporting": Object {
                "enabled": true,
              },
              "legend": Object {
                "itemStyle": Object {
                  "color": "white",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "10px",
                },
              },
              "subtitle": Object {
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "18px",
                },
                "text": "",
              },
              "title": Object {
                "style": Object {
                  "color": "#dfe6e9",
                  "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                  "fontSize": "18px",
                },
                "text": "Categorized, four series",
              },
              "tooltip": Object {
                "shared": true,
              },
              "yAxis": Object {
                "labels": Object {},
                "title": Object {
                  "style": Object {
                    "color": "#ffffff",
                    "fontFamily": "\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Verdana, Arial, Helvetica, sans-serif",
                    "fontSize": "12px",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                },
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
